tensorpack/models/conv2d.py

```diff
@@ -7,7 +7,7 @@
 from .common import layer_register, VariableHolder, rename_get_variable
 from ..utils.argtools import shape2d, shape4d
 
-__all__ = ['Conv2D', 'Deconv2D']
+__all__ = ['Conv2D', 'Deconv2D', 'AtrousConv2D']
```

Comment:
We've decided to not add any more layers to tensorpack. Because tensorflow has done the same thing already (`tf.layers`) and will probably maintain it better. See https://github.com/ppwwyyxx/tensorpack/issues/291.

Reply:
got it

Reply:
1. bn/mean/EMA, bn/variance/EMA cannot load from dict.

2.in tf.layers, kernel,bias is correspond to W,b in tensorpack.

group{}/block{}/conv{}/kernel:0    
group{}/block{}/conv{}/bias:0 

related log is as follows:

```
[0105 14:48:46 @sessinit.py:206] Variables to restore from dict: group2/block9/conv3/bn/beta:0, group2/block11/conv1/bn/variance/EMA:0, group0/block1/conv1/bn/beta:0, group2/block11/conv3/bn/gamma:0, group3/block0/conv3/W:0, group2/block17/conv1/bn/mean/EMA:0, group0/block0/conv3/bn/beta:0, group2/block15/conv1/bn/beta:0, group0/block2/conv3/W:0, group2/block10/conv1/bn/gamma:0, group2/block22/conv1/bn/gamma:0, group2/block6/conv3/bn/beta:0, group2/block3/conv3/bn/variance/EMA:0, group3/block1/conv1/bn/variance/EMA:0, group2/block1/conv1/bn/variance/EMA:0, group3/block1/conv1/W:0, group2/block15/conv3/bn/variance/EMA:0, group2/block14/conv1/bn/gamma:0, group2/block4/conv3/bn/beta:0, group3/block1/conv3/bn/mean/EMA:0, group2/block13/conv3/bn/gamma:0, group0/block0/conv1/bn/variance/EMA:0, group2/block14/conv3/bn/variance/EMA:0, group2/block15/conv1/W:0, group2/block20/conv3/W:0, group2/block18/conv3/bn/variance/EMA:0, group2/block22/conv1/bn/beta:0, group2/block0/convshortcut/bn/beta:0, group1/block2/conv3/W:0, group2/block19/conv3/bn/mean/EMA:0, group0/block0/convshortcut/bn/beta:0, group2/block15/conv1/bn/mean/EMA:0, group0/block1/conv3/bn/variance/EMA:0, group2/block21/conv3/bn/variance/EMA:0, group3/block1/conv3/bn/gamma:0, group3/block2/conv1/bn/mean/EMA:0, group2/block5/conv3/W:0, group2/block16/conv1/bn/mean/EMA:0, group2/block4/conv1/bn/mean/EMA:0, group2/block6/conv1/bn/variance/EMA:0, group2/block3/conv3/bn/beta:0, group1/block0/conv1/bn/gamma:0, group2/block7/conv1/bn/mean/EMA:0, group2/block3/conv1/W:0, group2/block22/conv1/bn/mean/EMA:0, group2/block10/conv1/bn/variance/EMA:0, group2/block21/conv3/bn/gamma:0, group2/block11/conv1/W:0, group2/block19/conv1/bn/beta:0, group2/block4/conv1/W:0, group2/block12/conv3/bn/gamma:0, group3/block2/conv1/W:0, group3/block1/conv1/bn/gamma:0, group2/block10/conv1/bn/beta:0, group2/block11/conv3/bn/variance/EMA:0, group2/block10/conv3/bn/beta:0, group1/block2/conv3/bn/mean/EMA:0, group0/block2/conv3/bn/beta:0, group2/block4/conv3/W:0, group2/block12/conv3/W:0, group2/block11/conv1/bn/beta:0, group2/block1/conv1/bn/beta:0, group2/block6/conv1/W:0, group2/block13/conv3/W:0, group2/block0/conv3/bn/gamma:0, group2/block9/conv1/W:0, group1/block3/conv3/bn/variance/EMA:0, conv0/bn/mean/EMA:0, group1/block3/conv1/bn/variance/EMA:0, group2/block4/conv3/bn/mean/EMA:0, group1/block0/convshortcut/bn/variance/EMA:0, group2/block3/conv1/bn/variance/EMA:0, group2/block21/conv1/W:0, group2/block9/conv1/bn/gamma:0, group3/block2/conv1/bn/variance/EMA:0, group3/block0/convshortcut/bn/variance/EMA:0, group2/block14/conv3/W:0, group1/block0/convshortcut/bn/beta:0, group1/block1/conv1/bn/mean/EMA:0, group2/block19/conv3/bn/variance/EMA:0, group2/block9/conv1/bn/mean/EMA:0, group2/block17/conv3/bn/gamma:0, group1/block1/conv3/bn/gamma:0, group0/block2/conv1/bn/beta:0, group2/block16/conv3/bn/beta:0, group1/block2/conv3/bn/variance/EMA:0, group1/block3/conv3/bn/mean/EMA:0, group2/block10/conv3/W:0, group0/block1/conv3/bn/beta:0, group2/block20/conv3/bn/mean/EMA:0, group3/block2/conv1/bn/gamma:0, group2/block9/conv1/bn/variance/EMA:0, group2/block22/conv1/bn/variance/EMA:0, group2/block17/conv3/bn/variance/EMA:0, group2/block11/conv3/W:0, group2/block11/conv1/bn/mean/EMA:0, group0/block0/convshortcut/W:0, group2/block12/conv3/bn/mean/EMA:0, group2/block3/conv1/bn/beta:0, group2/block0/convshortcut/bn/variance/EMA:0, group2/block0/conv3/bn/mean/EMA:0, group2/block19/conv1/bn/mean/EMA:0, group2/block14/conv3/bn/mean/EMA:0, group2/block15/conv1/bn/gamma:0, group1/block1/conv3/bn/beta:0, group2/block19/conv3/W:0, group2/block4/conv3/bn/gamma:0, group1/block2/conv1/bn/mean/EMA:0, group2/block21/conv1/bn/mean/EMA:0, group2/block22/conv3/W:0, group2/block18/conv3/bn/gamma:0, group2/block16/conv1/W:0, group2/block2/conv3/bn/variance/EMA:0, group2/block18/conv1/W:0, group2/block17/conv1/bn/variance/EMA:0, group2/block5/conv3/bn/gamma:0, group2/block10/conv1/W:0, group2/block16/conv3/bn/mean/EMA:0, group2/block5/conv1/bn/gamma:0, group2/block10/conv1/bn/mean/EMA:0, group2/block8/conv3/bn/beta:0, group2/block9/conv3/W:0, group2/block12/conv3/bn/variance/EMA:0, group1/block0/convshortcut/bn/gamma:0, group3/block0/conv3/bn/gamma:0, group2/block7/conv3/bn/mean/EMA:0, group0/block1/conv1/W:0, group1/block0/conv3/bn/mean/EMA:0, group2/block6/conv3/W:0, group2/block16/conv1/bn/gamma:0, group2/block0/conv1/bn/mean/EMA:0, group2/block5/conv1/bn/variance/EMA:0, group1/block0/conv1/bn/mean/EMA:0, group1/block2/conv1/bn/beta:0, group2/block21/conv1/bn/variance/EMA:0, group3/block2/conv3/bn/gamma:0, group3/block2/conv3/bn/beta:0, group2/block20/conv1/bn/beta:0, group2/block22/conv3/bn/mean/EMA:0, group2/block8/conv3/bn/mean/EMA:0, group2/block14/conv3/bn/beta:0, group2/block18/conv1/bn/mean/EMA:0, group2/block16/conv3/bn/gamma:0, group2/block10/conv3/bn/gamma:0, group1/block3/conv3/bn/gamma:0, group2/block21/conv3/bn/mean/EMA:0, group0/block2/conv1/bn/variance/EMA:0, group0/block2/conv1/bn/mean/EMA:0, group1/block0/convshortcut/bn/mean/EMA:0, group1/block0/conv1/bn/variance/EMA:0, group2/block6/conv3/bn/variance/EMA:0, group0/block1/conv3/W:0, group2/block11/conv3/bn/beta:0, group2/block13/conv1/bn/beta:0, group2/block4/conv1/bn/variance/EMA:0, group2/block5/conv3/bn/beta:0, group3/block0/conv3/bn/variance/EMA:0, group0/block2/conv1/W:0, group2/block2/conv1/bn/gamma:0, group2/block18/conv1/bn/variance/EMA:0, group1/block1/conv1/bn/beta:0, group2/block2/conv1/W:0, group2/block0/conv1/W:0, group2/block13/conv1/W:0, group2/block20/conv1/bn/gamma:0, group3/block1/conv1/bn/mean/EMA:0, group1/block2/conv3/bn/beta:0, group2/block17/conv3/W:0, group3/block0/convshortcut/bn/mean/EMA:0, group2/block21/conv3/bn/beta:0, group3/block1/conv1/bn/beta:0, group3/block0/convshortcut/bn/beta:0, group2/block17/conv1/W:0, group2/block3/conv1/bn/gamma:0, group2/block7/conv1/bn/beta:0, group2/block2/conv1/bn/mean/EMA:0, group2/block15/conv1/bn/variance/EMA:0, group0/block2/conv3/bn/gamma:0, group2/block8/conv3/bn/variance/EMA:0, group2/block4/conv3/bn/variance/EMA:0, group2/block20/conv3/bn/variance/EMA:0, group3/block0/conv3/bn/mean/EMA:0, group3/block0/convshortcut/bn/gamma:0, group0/block0/conv1/bn/mean/EMA:0, group2/block17/conv3/bn/beta:0, group0/block0/convshortcut/bn/gamma:0, group2/block18/conv1/bn/gamma:0, group2/block0/conv1/bn/gamma:0, group2/block2/conv3/bn/beta:0, conv0/W:0, group1/block0/convshortcut/W:0, group2/block9/conv3/bn/variance/EMA:0, group2/block0/conv3/W:0, group2/block8/conv1/W:0, group2/block7/conv1/bn/gamma:0, group2/block14/conv1/W:0, group3/block0/convshortcut/W:0, group1/block1/conv1/W:0, group2/block9/conv1/bn/beta:0, group1/block3/conv1/bn/mean/EMA:0, group2/block7/conv3/W:0, group2/block5/conv1/bn/mean/EMA:0, group3/block2/conv3/W:0, group2/block0/convshortcut/W:0, group2/block1/conv3/bn/gamma:0, group1/block0/conv3/bn/variance/EMA:0, group2/block16/conv1/bn/variance/EMA:0, group2/block21/conv1/bn/beta:0, group2/block2/conv3/W:0, group2/block3/conv3/W:0, group0/block0/conv3/bn/mean/EMA:0, group3/block0/conv1/bn/mean/EMA:0, group1/block0/conv3/bn/gamma:0, group2/block7/conv1/W:0, group2/block0/conv1/bn/variance/EMA:0, group1/block2/conv3/bn/gamma:0, group2/block2/conv3/bn/gamma:0, group2/block22/conv3/bn/variance/EMA:0, group2/block8/conv1/bn/mean/EMA:0, group2/block12/conv3/bn/beta:0, group2/block15/conv3/bn/gamma:0, group1/block0/conv3/bn/beta:0, group2/block22/conv3/bn/beta:0, conv0/bn/beta:0, group3/block1/conv3/bn/beta:0, conv0/bn/variance/EMA:0, group3/block0/conv3/bn/beta:0, group2/block15/conv3/bn/beta:0, group2/block13/conv3/bn/variance/EMA:0, group0/block2/conv3/bn/mean/EMA:0, group1/block1/conv3/W:0, group3/block0/conv1/bn/beta:0, group2/block16/conv1/bn/beta:0, group1/block1/conv1/bn/gamma:0, group2/block10/conv3/bn/variance/EMA:0, group2/block6/conv1/bn/gamma:0, group0/block0/conv3/bn/gamma:0, group2/block12/conv1/bn/gamma:0, group2/block0/conv3/bn/beta:0, group1/block3/conv3/bn/beta:0, group0/block0/convshortcut/bn/variance/EMA:0, group0/block0/convshortcut/bn/mean/EMA:0, group2/block1/conv3/bn/variance/EMA:0, group2/block16/conv3/bn/variance/EMA:0, group2/block6/conv1/bn/beta:0, group1/block1/conv3/bn/variance/EMA:0, group2/block15/conv3/W:0, group2/block1/conv1/bn/gamma:0, group2/block4/conv1/bn/beta:0, group0/block0/conv3/W:0, group3/block2/conv3/bn/variance/EMA:0, group2/block17/conv1/bn/gamma:0, group1/block3/conv3/W:0, group2/block1/conv1/bn/mean/EMA:0, group2/block8/conv1/bn/gamma:0, group2/block12/conv1/bn/mean/EMA:0, group3/block1/conv3/W:0, group3/block2/conv3/bn/mean/EMA:0, group2/block2/conv3/bn/mean/EMA:0, group2/block19/conv3/bn/beta:0, group2/block6/conv3/bn/gamma:0, group2/block22/conv1/W:0, group2/block18/conv3/bn/mean/EMA:0, group2/block14/conv3/bn/gamma:0, group2/block18/conv1/bn/beta:0, group0/block1/conv1/bn/variance/EMA:0, group1/block3/conv1/bn/gamma:0, group2/block12/conv1/W:0, group0/block1/conv1/bn/gamma:0, group2/block11/conv3/bn/mean/EMA:0, group2/block12/conv1/bn/beta:0, group2/block5/conv1/W:0, group2/block0/conv1/bn/beta:0, group1/block2/conv1/bn/gamma:0, group2/block20/conv3/bn/beta:0, group2/block13/conv1/bn/mean/EMA:0, group1/block3/conv1/bn/beta:0, group2/block1/conv3/W:0, group3/block0/conv1/bn/gamma:0, group2/block8/conv3/bn/gamma:0, group0/block1/conv3/bn/mean/EMA:0, group1/block2/conv1/W:0, group1/block1/conv1/bn/variance/EMA:0, group2/block2/conv1/bn/beta:0, group2/block4/conv1/bn/gamma:0, group2/block14/conv1/bn/mean/EMA:0, group0/block0/conv1/W:0, group1/block1/conv3/bn/mean/EMA:0, conv0/bn/gamma:0, group2/block1/conv3/bn/mean/EMA:0, group2/block7/conv1/bn/variance/EMA:0, group2/block0/conv3/bn/variance/EMA:0, group2/block20/conv3/bn/gamma:0, group3/block0/conv1/W:0, group2/block11/conv1/bn/gamma:0, group2/block13/conv3/bn/beta:0, group2/block19/conv3/bn/gamma:0, group2/block7/conv3/bn/gamma:0, group2/block14/conv1/bn/variance/EMA:0, group2/block1/conv1/W:0, group2/block6/conv1/bn/mean/EMA:0, group2/block0/convshortcut/bn/gamma:0, group0/block1/conv1/bn/mean/EMA:0, group2/block17/conv1/bn/beta:0, group0/block0/conv3/bn/variance/EMA:0, group2/block14/conv1/bn/beta:0, group2/block8/conv1/bn/variance/EMA:0, group2/block20/conv1/bn/variance/EMA:0, group2/block5/conv3/bn/mean/EMA:0, group2/block18/conv3/W:0, group2/block8/conv1/bn/beta:0, group2/block18/conv3/bn/beta:0, group2/block2/conv1/bn/variance/EMA:0, group0/block1/conv3/bn/gamma:0, group1/block0/conv3/W:0, group2/block7/conv3/bn/variance/EMA:0, group2/block20/conv1/W:0, group0/block2/conv3/bn/variance/EMA:0, group2/block13/conv1/bn/variance/EMA:0, group3/block0/conv1/bn/variance/EMA:0, group2/block13/conv1/bn/gamma:0, group2/block5/conv1/bn/beta:0, group2/block21/conv3/W:0, group2/block7/conv3/bn/beta:0, group2/block20/conv1/bn/mean/EMA:0, group0/block0/conv1/bn/beta:0, group2/block17/conv3/bn/mean/EMA:0, group0/block0/conv1/bn/gamma:0, group2/block0/convshortcut/bn/mean/EMA:0, group2/block10/conv3/bn/mean/EMA:0, group0/block2/conv1/bn/gamma:0, group2/block15/conv3/bn/mean/EMA:0, group2/block1/conv3/bn/beta:0, group2/block6/conv3/bn/mean/EMA:0, group2/block12/conv1/bn/variance/EMA:0, group2/block9/conv3/bn/gamma:0, group2/block13/conv3/bn/mean/EMA:0, group3/block1/conv3/bn/variance/EMA:0, group2/block16/conv3/W:0, group3/block2/conv1/bn/beta:0, group2/block3/conv1/bn/mean/EMA:0, group1/block0/conv1/bn/beta:0, group2/block19/conv1/bn/variance/EMA:0, group2/block22/conv3/bn/gamma:0, group1/block0/conv1/W:0, group2/block5/conv3/bn/variance/EMA:0, group1/block2/conv1/bn/variance/EMA:0, group2/block3/conv3/bn/mean/EMA:0, group1/block3/conv1/W:0, group2/block9/conv3/bn/mean/EMA:0, group2/block19/conv1/bn/gamma:0, group2/block19/conv1/W:0, group2/block21/conv1/bn/gamma:0, group2/block8/conv3/W:0, group2/block3/conv3/bn/gamma:0
[0105 14:48:46 @sessinit.py:89] WRN The following variables are in the graph, but not found in the dict: aspp6_conv/bias:0, aspp6_conv/kernel:0, beta1_power:0, beta2_power:0, bn/beta:0, bn/gamma:0, bn/mean/EMA:0, bn/variance/EMA:0, global_step:0, group0/block0/bn/beta:0, group0/block0/bn/gamma:0, group0/block0/bn/mean/EMA:0, group0/block0/bn/variance/EMA:0, group0/block0/conv2/bias:0, group0/block0/conv2/kernel:0, group0/block1/bn/beta:0, group0/block1/bn/gamma:0, group0/block1/bn/mean/EMA:0, group0/block1/bn/variance/EMA:0, group0/block1/conv2/bias:0, group0/block1/conv2/kernel:0, group0/block2/bn/beta:0, group0/block2/bn/gamma:0, group0/block2/bn/mean/EMA:0, group0/block2/bn/variance/EMA:0, group0/block2/conv2/bias:0, group0/block2/conv2/kernel:0, group1/block0/bn/beta:0, group1/block0/bn/gamma:0, group1/block0/bn/mean/EMA:0, group1/block0/bn/variance/EMA:0, group1/block0/conv2/bias:0, group1/block0/conv2/kernel:0, group1/block1/bn/beta:0, group1/block1/bn/gamma:0, group1/block1/bn/mean/EMA:0, group1/block1/bn/variance/EMA:0, group1/block1/conv2/bias:0, group1/block1/conv2/kernel:0, group1/block2/bn/beta:0, group1/block2/bn/gamma:0, group1/block2/bn/mean/EMA:0, group1/block2/bn/variance/EMA:0, group1/block2/conv2/bias:0, group1/block2/conv2/kernel:0, group1/block3/bn/beta:0, group1/block3/bn/gamma:0, group1/block3/bn/mean/EMA:0, group1/block3/bn/variance/EMA:0, group1/block3/conv2/bias:0, group1/block3/conv2/kernel:0, group2/block0/bn/beta:0, group2/block0/bn/gamma:0, group2/block0/bn/mean/EMA:0, group2/block0/bn/variance/EMA:0, group2/block0/conv2/bias:0, group2/block0/conv2/kernel:0, group2/block1/bn/beta:0, group2/block1/bn/gamma:0, group2/block1/bn/mean/EMA:0, group2/block1/bn/variance/EMA:0, group2/block1/conv2/bias:0, group2/block1/conv2/kernel:0, group2/block10/bn/beta:0, group2/block10/bn/gamma:0, group2/block10/bn/mean/EMA:0, group2/block10/bn/variance/EMA:0, group2/block10/conv2/bias:0, group2/block10/conv2/kernel:0, group2/block11/bn/beta:0, group2/block11/bn/gamma:0, group2/block11/bn/mean/EMA:0, group2/block11/bn/variance/EMA:0, group2/block11/conv2/bias:0, group2/block11/conv2/kernel:0, group2/block12/bn/beta:0, group2/block12/bn/gamma:0, group2/block12/bn/mean/EMA:0, group2/block12/bn/variance/EMA:0, group2/block12/conv2/bias:0, group2/block12/conv2/kernel:0, group2/block13/bn/beta:0, group2/block13/bn/gamma:0, group2/block13/bn/mean/EMA:0, group2/block13/bn/variance/EMA:0, group2/block13/conv2/bias:0, group2/block13/conv2/kernel:0, group2/block14/bn/beta:0, group2/block14/bn/gamma:0, group2/block14/bn/mean/EMA:0, group2/block14/bn/variance/EMA:0, group2/block14/conv2/bias:0, group2/block14/conv2/kernel:0, group2/block15/bn/beta:0, group2/block15/bn/gamma:0, group2/block15/bn/mean/EMA:0, group2/block15/bn/variance/EMA:0, group2/block15/conv2/bias:0, group2/block15/conv2/kernel:0, group2/block16/bn/beta:0, group2/block16/bn/gamma:0, group2/block16/bn/mean/EMA:0, group2/block16/bn/variance/EMA:0, group2/block16/conv2/bias:0, group2/block16/conv2/kernel:0, group2/block17/bn/beta:0, group2/block17/bn/gamma:0, group2/block17/bn/mean/EMA:0, group2/block17/bn/variance/EMA:0, group2/block17/conv2/bias:0, group2/block17/conv2/kernel:0, group2/block18/bn/beta:0, group2/block18/bn/gamma:0, group2/block18/bn/mean/EMA:0, group2/block18/bn/variance/EMA:0, group2/block18/conv2/bias:0, group2/block18/conv2/kernel:0, group2/block19/bn/beta:0, group2/block19/bn/gamma:0, group2/block19/bn/mean/EMA:0, group2/block19/bn/variance/EMA:0, group2/block19/conv2/bias:0, group2/block19/conv2/kernel:0, group2/block2/bn/beta:0, group2/block2/bn/gamma:0, group2/block2/bn/mean/EMA:0, group2/block2/bn/variance/EMA:0, group2/block2/conv2/bias:0, group2/block2/conv2/kernel:0, group2/block20/bn/beta:0, group2/block20/bn/gamma:0, group2/block20/bn/mean/EMA:0, group2/block20/bn/variance/EMA:0, group2/block20/conv2/bias:0, group2/block20/conv2/kernel:0, group2/block21/bn/beta:0, group2/block21/bn/gamma:0, group2/block21/bn/mean/EMA:0, group2/block21/bn/variance/EMA:0, group2/block21/conv2/bias:0, group2/block21/conv2/kernel:0, group2/block22/bn/beta:0, group2/block22/bn/gamma:0, group2/block22/bn/mean/EMA:0, group2/block22/bn/variance/EMA:0, group2/block22/conv2/bias:0, group2/block22/conv2/kernel:0, group2/block3/bn/beta:0, group2/block3/bn/gamma:0, group2/block3/bn/mean/EMA:0, group2/block3/bn/variance/EMA:0, group2/block3/conv2/bias:0, group2/block3/conv2/kernel:0, group2/block4/bn/beta:0, group2/block4/bn/gamma:0, group2/block4/bn/mean/EMA:0, group2/block4/bn/variance/EMA:0, group2/block4/conv2/bias:0, group2/block4/conv2/kernel:0, group2/block5/bn/beta:0, group2/block5/bn/gamma:0, group2/block5/bn/mean/EMA:0, group2/block5/bn/variance/EMA:0, group2/block5/conv2/bias:0, group2/block5/conv2/kernel:0, group2/block6/bn/beta:0, group2/block6/bn/gamma:0, group2/block6/bn/mean/EMA:0, group2/block6/bn/variance/EMA:0, group2/block6/conv2/bias:0, group2/block6/conv2/kernel:0, group2/block7/bn/beta:0, group2/block7/bn/gamma:0, group2/block7/bn/mean/EMA:0, group2/block7/bn/variance/EMA:0, group2/block7/conv2/bias:0, group2/block7/conv2/kernel:0, group2/block8/bn/beta:0, group2/block8/bn/gamma:0, group2/block8/bn/mean/EMA:0, group2/block8/bn/variance/EMA:0, group2/block8/conv2/bias:0, group2/block8/conv2/kernel:0, group2/block9/bn/beta:0, group2/block9/bn/gamma:0, group2/block9/bn/mean/EMA:0, group2/block9/bn/variance/EMA:0, group2/block9/conv2/bias:0, group2/block9/conv2/kernel:0, group3/block0/bn/beta:0, group3/block0/bn/gamma:0, group3/block0/bn/mean/EMA:0, group3/block0/bn/variance/EMA:0, group3/block0/conv2/bias:0, group3/block0/conv2/kernel:0, group3/block1/bn/beta:0, group3/block1/bn/gamma:0, group3/block1/bn/mean/EMA:0, group3/block1/bn/variance/EMA:0, group3/block1/conv2/bias:0, group3/block1/conv2/kernel:0, group3/block2/bn/beta:0, group3/block2/bn/gamma:0, group3/block2/bn/mean/EMA:0, group3/block2/bn/variance/EMA:0, group3/block2/conv2/bias:0, group3/block2/conv2/kernel:0, learning_rate:0
[0105 14:48:46 @sessinit.py:89] WRN The following variables are in the dict, but not found in the graph: group0/block0/conv2/W:0, group0/block0/conv2/bn/beta:0, group0/block0/conv2/bn/gamma:0, group0/block0/conv2/bn/mean/EMA:0, group0/block0/conv2/bn/variance/EMA:0, group0/block1/conv2/W:0, group0/block1/conv2/bn/beta:0, group0/block1/conv2/bn/gamma:0, group0/block1/conv2/bn/mean/EMA:0, group0/block1/conv2/bn/variance/EMA:0, group0/block2/conv2/W:0, group0/block2/conv2/bn/beta:0, group0/block2/conv2/bn/gamma:0, group0/block2/conv2/bn/mean/EMA:0, group0/block2/conv2/bn/variance/EMA:0, group1/block0/conv2/W:0, group1/block0/conv2/bn/beta:0, group1/block0/conv2/bn/gamma:0, group1/block0/conv2/bn/mean/EMA:0, group1/block0/conv2/bn/variance/EMA:0, group1/block1/conv2/W:0, group1/block1/conv2/bn/beta:0, group1/block1/conv2/bn/gamma:0, group1/block1/conv2/bn/mean/EMA:0, group1/block1/conv2/bn/variance/EMA:0, group1/block2/conv2/W:0, group1/block2/conv2/bn/beta:0, group1/block2/conv2/bn/gamma:0, group1/block2/conv2/bn/mean/EMA:0, group1/block2/conv2/bn/variance/EMA:0, group1/block3/conv2/W:0, group1/block3/conv2/bn/beta:0, group1/block3/conv2/bn/gamma:0, group1/block3/conv2/bn/mean/EMA:0, group1/block3/conv2/bn/variance/EMA:0, group2/block0/conv2/W:0, group2/block0/conv2/bn/beta:0, group2/block0/conv2/bn/gamma:0, group2/block0/conv2/bn/mean/EMA:0, group2/block0/conv2/bn/variance/EMA:0, group2/block1/conv2/W:0, group2/block1/conv2/bn/beta:0, group2/block1/conv2/bn/gamma:0, group2/block1/conv2/bn/mean/EMA:0, group2/block1/conv2/bn/variance/EMA:0, group2/block10/conv2/W:0, group2/block10/conv2/bn/beta:0, group2/block10/conv2/bn/gamma:0, group2/block10/conv2/bn/mean/EMA:0, group2/block10/conv2/bn/variance/EMA:0, group2/block11/conv2/W:0, group2/block11/conv2/bn/beta:0, group2/block11/conv2/bn/gamma:0, group2/block11/conv2/bn/mean/EMA:0, group2/block11/conv2/bn/variance/EMA:0, group2/block12/conv2/W:0, group2/block12/conv2/bn/beta:0, group2/block12/conv2/bn/gamma:0, group2/block12/conv2/bn/mean/EMA:0, group2/block12/conv2/bn/variance/EMA:0, group2/block13/conv2/W:0, group2/block13/conv2/bn/beta:0, group2/block13/conv2/bn/gamma:0, group2/block13/conv2/bn/mean/EMA:0, group2/block13/conv2/bn/variance/EMA:0, group2/block14/conv2/W:0, group2/block14/conv2/bn/beta:0, group2/block14/conv2/bn/gamma:0, group2/block14/conv2/bn/mean/EMA:0, group2/block14/conv2/bn/variance/EMA:0, group2/block15/conv2/W:0, group2/block15/conv2/bn/beta:0, group2/block15/conv2/bn/gamma:0, group2/block15/conv2/bn/mean/EMA:0, group2/block15/conv2/bn/variance/EMA:0, group2/block16/conv2/W:0, group2/block16/conv2/bn/beta:0, group2/block16/conv2/bn/gamma:0, group2/block16/conv2/bn/mean/EMA:0, group2/block16/conv2/bn/variance/EMA:0, group2/block17/conv2/W:0, group2/block17/conv2/bn/beta:0, group2/block17/conv2/bn/gamma:0, group2/block17/conv2/bn/mean/EMA:0, group2/block17/conv2/bn/variance/EMA:0, group2/block18/conv2/W:0, group2/block18/conv2/bn/beta:0, group2/block18/conv2/bn/gamma:0, group2/block18/conv2/bn/mean/EMA:0, group2/block18/conv2/bn/variance/EMA:0, group2/block19/conv2/W:0, group2/block19/conv2/bn/beta:0, group2/block19/conv2/bn/gamma:0, group2/block19/conv2/bn/mean/EMA:0, group2/block19/conv2/bn/variance/EMA:0, group2/block2/conv2/W:0, group2/block2/conv2/bn/beta:0, group2/block2/conv2/bn/gamma:0, group2/block2/conv2/bn/mean/EMA:0, group2/block2/conv2/bn/variance/EMA:0, group2/block20/conv2/W:0, group2/block20/conv2/bn/beta:0, group2/block20/conv2/bn/gamma:0, group2/block20/conv2/bn/mean/EMA:0, group2/block20/conv2/bn/variance/EMA:0, group2/block21/conv2/W:0, group2/block21/conv2/bn/beta:0, group2/block21/conv2/bn/gamma:0, group2/block21/conv2/bn/mean/EMA:0, group2/block21/conv2/bn/variance/EMA:0, group2/block22/conv2/W:0, group2/block22/conv2/bn/beta:0, group2/block22/conv2/bn/gamma:0, group2/block22/conv2/bn/mean/EMA:0, group2/block22/conv2/bn/variance/EMA:0, group2/block3/conv2/W:0, group2/block3/conv2/bn/beta:0, group2/block3/conv2/bn/gamma:0, group2/block3/conv2/bn/mean/EMA:0, group2/block3/conv2/bn/variance/EMA:0, group2/block4/conv2/W:0, group2/block4/conv2/bn/beta:0, group2/block4/conv2/bn/gamma:0, group2/block4/conv2/bn/mean/EMA:0, group2/block4/conv2/bn/variance/EMA:0, group2/block5/conv2/W:0, group2/block5/conv2/bn/beta:0, group2/block5/conv2/bn/gamma:0, group2/block5/conv2/bn/mean/EMA:0, group2/block5/conv2/bn/variance/EMA:0, group2/block6/conv2/W:0, group2/block6/conv2/bn/beta:0, group2/block6/conv2/bn/gamma:0, group2/block6/conv2/bn/mean/EMA:0, group2/block6/conv2/bn/variance/EMA:0, group2/block7/conv2/W:0, group2/block7/conv2/bn/beta:0, group2/block7/conv2/bn/gamma:0, group2/block7/conv2/bn/mean/EMA:0, group2/block7/conv2/bn/variance/EMA:0, group2/block8/conv2/W:0, group2/block8/conv2/bn/beta:0, group2/block8/conv2/bn/gamma:0, group2/block8/conv2/bn/mean/EMA:0, group2/block8/conv2/bn/variance/EMA:0, group2/block9/conv2/W:0, group2/block9/conv2/bn/beta:0, group2/block9/conv2/bn/gamma:0, group2/block9/conv2/bn/mean/EMA:0, group2/block9/conv2/bn/variance/EMA:0, group3/block0/conv2/W:0, group3/block0/conv2/bn/beta:0, group3/block0/conv2/bn/gamma:0, group3/block0/conv2/bn/mean/EMA:0, group3/block0/conv2/bn/variance/EMA:0, group3/block1/conv2/W:0, group3/block1/conv2/bn/beta:0, group3/block1/conv2/bn/gamma:0, group3/block1/conv2/bn/mean/EMA:0, group3/block1/conv2/bn/variance/EMA:0, group3/block2/conv2/W:0, group3/block2/conv2/bn/beta:0, group3/block2/conv2/bn/gamma:0, group3/block2/conv2/bn/mean/EMA:0, group3/block2/conv2/bn/variance/EMA:0, linear/W:0, linear/b:
```

Reply:
final solution:   I move the atrous convolution implementation into deeplab example, so that it will not pollute the tensorpack kernel.